File: VB/WindowsApplication3/OfficeNavigationBarHelper.vb
Imports System
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.Utils
Imports DevExpress.XtraBars.Navigation

Namespace DxExample
    Public Class OfficeNavigationBarHelper
        Private _timer As Timer
        Private _currentPeekForm As FlyoutPanelBeakForm
        Private _officeNavigationBar As OfficeNavigationBar
        Public Sub New(ByVal _officeNavigationBar As OfficeNavigationBar)
            Me._officeNavigationBar = _officeNavigationBar
        End Sub
        Private ReadOnly Property Timer() As Timer
            Get
                If _timer Is Nothing Then
                    SetUpTimer()
                End If
                Return _timer
            End Get
        End Property
        Private Sub SetUpTimer()
            _timer = New Timer()
            _timer.Interval = 200
            AddHandler _timer.Tick, AddressOf timer_Tick
        End Sub
        Private Sub timer_Tick(ByVal sender As Object, ByVal e As EventArgs)
            Dim point As Point = Control.MousePosition
            If CanHidePeekForm Then
                _officeNavigationBar.HidePeekForm()
                Timer.Stop()
            End If
        End Sub
        Private ReadOnly Property CanHidePeekForm() As Boolean
            Get
                Dim screenPoint As Point = Control.MousePosition
                Dim clientPoint As Point = _officeNavigationBar.Parent.PointToClient(Control.MousePosition)
                Return (Not _currentPeekForm.Bounds.Contains(screenPoint)) AndAlso Not _officeNavigationBar.Bounds.Contains(clientPoint)
            End Get
        End Property
        Public Sub SetPeekFormBehavior(ByVal autoHideOnMouseLeave As Boolean)
            RemoveHandler _officeNavigationBar.PeekFormShown, AddressOf OnPeekFormShown
            If _timer IsNot Nothing Then
                RemoveHandler _timer.Tick, AddressOf timer_Tick
                _timer.Dispose()
                _timer = Nothing
            End If
            If autoHideOnMouseLeave Then
                AddHandler _officeNavigationBar.PeekFormShown, AddressOf OnPeekFormShown
            End If
        End Sub
        Private Sub OnPeekFormShown(ByVal sender As Object, ByVal e As DevExpress.XtraBars.Navigation.NavigationPeekFormEventArgs)
            _currentPeekForm = TryCast(e.Control.FindForm(), FlyoutPanelBeakForm)
            Timer.Stop()
            Timer.Start()
        End Sub
    End Class
End Namespace

File: CS/WindowsApplication3/OfficeNavigationBarHelper.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.Utils;
using DevExpress.XtraBars.Navigation;

namespace DxExample {
    public class OfficeNavigationBarHelper {
        Timer _timer;
        FlyoutPanelBeakForm _currentPeekForm;
        OfficeNavigationBar _officeNavigationBar;
        public OfficeNavigationBarHelper(OfficeNavigationBar _officeNavigationBar) {
            this._officeNavigationBar = _officeNavigationBar;
        }
        Timer Timer {
            get {
                if (_timer == null)
                    SetUpTimer();
                return _timer;
            }
        }
        void SetUpTimer() {
            _timer = new Timer();
            _timer.Interval = 200;
            _timer.Tick += timer_Tick;
        }
        void timer_Tick(object sender, EventArgs e) {
            Point point = Control.MousePosition;
            if (CanHidePeekForm) {
                _officeNavigationBar.HidePeekForm();
                Timer.Stop();
            }
        }
        bool CanHidePeekForm {
            get {
                Point screenPoint = Control.MousePosition;
                Point clientPoint = _officeNavigationBar.Parent.PointToClient(Control.MousePosition);
                return !_currentPeekForm.Bounds.Contains(screenPoint) && !_officeNavigationBar.Bounds.Contains(clientPoint);
            }
        }
        public void SetPeekFormBehavior(bool autoHideOnMouseLeave) {
            _officeNavigationBar.PeekFormShown -= OnPeekFormShown;
            if (_timer != null) {
                _timer.Tick -= timer_Tick;
                _timer.Dispose();
                _timer = null;
            }
            if (autoHideOnMouseLeave)
                _officeNavigationBar.PeekFormShown += OnPeekFormShown;
        }
        void OnPeekFormShown(object sender, DevExpress.XtraBars.Navigation.NavigationPeekFormEventArgs e) {
            _currentPeekForm = e.Control.FindForm() as FlyoutPanelBeakForm;
            Timer.Stop();
            Timer.Start();
        }
    }
}
